libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -684,6 +684,67 @@ impl Greenlight {
         _ = self.inprogress_payments.fetch_sub(1, Ordering::Relaxed);
         res
     }
+
+    async fn wait_channel_reestablished(&self, path: &PaymentPath) -> NodeResult<()> {
+        let deadline =
+            Instant::now()
+                .checked_add(Duration::from_secs(20))
+                .ok_or(NodeError::generic(
+                    "Failed to set channel establishment deadline",
+                ))?;
+
+        while Instant::now().le(&deadline) && !self.poll_channel_reestablished(path).await? {
+            tokio::time::sleep(Duration::from_millis(50)).await
+        }
+
+        Ok(())
```

Comment:
It looks like if we pass the deadline without channel reestablish the send_pay is still attempted, is that intended?

Reply:
No, that was not intended, good catch!
However, now that you mention it, it might actually be fine to attempt that payment here.

Reply:
I don't have a problem with that